samcli/lib/telemetry/metric.py

```diff
@@ -476,6 +479,28 @@ def _get_execution_environment(self) -> str:
             return cicd_platform.name
         return "CLI"
 
+    def _get_docker_host(self) -> str:
+        """
+        Returns the last part of a DOCKER_HOST string. Has conditional logic to properly parse
+        URLs and local file system paths. If the DOCKER_HOST is not set, returns an empty string.
+
+        Examples:
+            - unix:///var/run/docker.sock -> docker.sock
+            - tcp://localhost:1234 -> localhost:1234
+            - /var/run/docker.sock -> docker.sock
+        """
+
+        parsed = urlparse(self._gc.docker_host)
+        if not parsed.scheme == "":
```

Comment:
Did you mean the opposite here? I imagine that **if** the schema is empty, then I can check if it's a file that exists, but here you're doing that if it's **not** empty. If it happens that it's not empty, then we could be calling `os.path.exists("unix:///var/run/docker.sock")` which returns False.